addons/point_of_sale/models/pos_order.py

```diff
@@ -889,14 +889,52 @@ def create_from_ui(self, orders, draft=False):
         """
         order_ids = []
         for order in orders:
-            existing_order = False
-            if 'server_id' in order['data']:
-                existing_order = self.env['pos.order'].search(['|', ('id', '=', order['data']['server_id']), ('pos_reference', '=', order['data']['name'])], limit=1)
-            if (existing_order and existing_order.state == 'draft') or not existing_order:
-                order_ids.append(self._process_order(order, draft, existing_order))
+            existing_draft_order = None
+
+            if 'server_id' in order['data'] and order['data']['server_id']:
+                # if the server id exists, it must only search based on the id
+                existing_draft_order = self.env['pos.order'].search(['&', ('id', '=', order['data']['server_id']), ('state', '=', 'draft')], limit=1)
+
+                # if there is no draft order, skip processing this order
+                if not existing_draft_order:
+                    continue
+
+            if not existing_draft_order:
+                existing_draft_order = self.env['pos.order'].search(['&', ('pos_reference', '=', order['data']['name']), ('state', '=', 'draft')], limit=1)
+
+            if existing_draft_order:
+                order_ids.append(self._process_order(order, draft, existing_draft_order))
+            else:
+                existing_orders = self.env['pos.order'].search([('pos_reference', '=', order['data']['name'])])
+                if all(not self._is_the_same_order(order['data'], existing_order) for existing_order in existing_orders):
+                    order_ids.append(self._process_order(order, draft, None))
```

Comment:
```suggestion
                    order_ids.append(self._process_order(order, draft, False))
```

It was originally passing `False`, not `None`.

Reply:
@caburj I changed it.